xlators/features/shard/src/shard.c

```diff
@@ -1917,16 +1917,22 @@ shard_stat(call_frame_t *frame, xlator_t *this, loc_t *loc, dict_t *xdata)
         return 0;
     }
 
-    ret = shard_inode_ctx_get_block_size(loc->inode, this, &block_size);
-    if (ret) {
-        gf_msg(this->name, GF_LOG_ERROR, 0, SHARD_MSG_INODE_CTX_GET_FAILED,
-               "Failed to get block "
-               "size from inode ctx of %s",
-               uuid_utoa(loc->inode->gfid));
-        goto err;
-    }
+    if (frame->root->pid != GF_CLIENT_PID_GSYNCD) {
```

Comment:
I'd do it differently:
if (frame->root->pid == GF_CLIENT_PID_GSYNC) {
    STACK_WIND(...)
    return;
}

...